### DISCUSSION TITLE
BMR Transport and RPC requirement

### DISCUSSION THREAD

**Discussion by mohamedelshami** [icon-project/btp#28] [Q&A] [answered]
Is there a requirement for maintaining a common [jsonrpc](https://github.com/icon-project/btp/blob/master/common/jsonrpc/client.go) for all chain clients?

We're currently leveraging BSC [rpc client](https://github.com/binance-chain/bsc/blob/master/rpc/client.go) to monitor block and handle subscriptions properly. Also makes it easy to maintain a consistent client interface with the target network.

Apart from from runtime dependency optimisation or increased binary size, I don't see this having an impact on behaviour of the BMR monitor/update process.

Also, we might need to introduce a second endpoint for ws protocol option in the BMR cmd (btpsimple.main) - Binance Smart Chain runs websocket on a different port.

Thoughts?

**Comment by jspark-icon** [comment, marked as answer]
There is no requirement for using `/common/jsonrpc`
For second endpoint or any other options, you could use [src.options or dst.options](https://github.com/icon-project/btp/blob/master/cmd/btpsimple/main.go#L155) as [module/icon/sender.opt](https://github.com/icon-project/btp/blob/master/cmd/btpsimple/module/icon/sender.go#L51)
